pkg/awsconfig/testing/firewallConfig.go

```diff
@@ -0,0 +1,479 @@
+	FirewallSubnet := awsClient.CreateSubnet("10.0.2.0/24", Vpc)
+	//Create PublicSubnet Route Table
+	PublicRT := awsClient.CreateRouteTableForSubnet(Vpc, PublicSubnet)
+	//Create PrivateSubnet Route Table
+	PrivateRT := awsClient.CreateRouteTableForSubnet(Vpc, PrivateSubnet)
+	//Create FirewallSubnet Route Table
+	FirewallRT := awsClient.CreateRouteTableForSubnet(Vpc, FirewallSubnet)
+	//Create IGW Route Table
+	IgRT := awsClient.CreateRouteTableForIGW(Vpc, IG)
+	//Create NAT Gateway
+	NatGateway := awsClient.CreateNatGateway(PublicSubnet)
+
+	//Create route 0.0.0.0/0 in PrivateRT for NatGateway
+	awsClient.CreateRouteForGateway("0.0.0.0/0", *NatGateway.NatGateway.NatGatewayId, PrivateRT)
+	fmt.Println("Successfully Created a route 0.0.0.0/0 to NatGateway in Private Subnet")
+	//Create route 0.0.0.0/0 in FirewallSubnet for IG
+	awsClient.CreateRouteForGateway("0.0.0.0/0", *IG.InternetGateway.InternetGatewayId, FirewallRT)
+	fmt.Println("Successfully Created a route 0.0.0.0/0 to IGW in Firewall Subnet")
+
+	//Create Firewall
+	Firewall := awsClient.CreateFirewall(FirewallSubnet, Vpc)
+	//Wait for 6 minutes for the Firewall to be ready
+	fmt.Println("It's gonna take around 6 minutes for the Firewall Vpc Endpoint to become available")
+	if awsClient.IsFirewallReady(Firewall) != nil {
+		fmt.Println(awsClient.IsFirewallReady(Firewall).Error())
+	}
+	fmt.Println("VpcEndpoint is Now Available!")
+
+	DescribeFirewall := awsClient.DescribeFirewall(Firewall)
+	//Create route 0.0.0.0/0 in PublicRT for FirewallEndpoint
+	firewallEndpointId := *DescribeFirewall.FirewallStatus.SyncStates[*FirewallSubnet.Subnet.AvailabilityZone].Attachment.EndpointId
+	//Check to see if the Firewall VpcEndpoint is available
+	awsClient.CreateRouteToFirewall("0.0.0.0/0", firewallEndpointId, PublicRT)
+	fmt.Println("Successfully route 0.0.0.0/0 to the Firewall Endpoint in PublicRT ")
+	//Create route 10.0.0.0/24 in IgRt to FirewallEndpoint
+	awsClient.CreateRouteToFirewall("10.0.0.0/24", firewallEndpointId, IgRT)
+	fmt.Println("Successfully route 10.0.0.0/24 to the Firewall Endpoint in IgRT ")
+
+}
+
+func (c Client) CreateVPC() ec2.CreateVpcOutput {
+	VPC, err := c.ec2Client.CreateVpc(&ec2.CreateVpcInput{
+		CidrBlock: aws.String("10.0.0.0/16"),
+	})
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+
+	fmt.Println("Successfully created a vpc with ID:", string(*VPC.Vpc.VpcId))
+	//Enable DNSHostname
+	_, err = c.ec2Client.ModifyVpcAttribute(&ec2.ModifyVpcAttributeInput{
+		EnableDnsHostnames: &ec2.AttributeBooleanValue{
+			Value: aws.Bool(true),
+		},
+		VpcId: aws.String(*VPC.Vpc.VpcId),
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				fmt.Println(aerr.Error())
+			}
+		} else {
+			fmt.Println(err.Error())
+		}
+	}
+	fmt.Println("Successfully enabled DNSHostname for the newly created VPC")
+	return *VPC
+}
+
+func (c Client) CreateInternetGateway(Vpc ec2.CreateVpcOutput) ec2.CreateInternetGatewayOutput {
```

Comment:
Given this function is attaching an IGW to a VPC, maybe the name should be `CreateInternetGatewayForVPC` and should just take a VPC ID as input.